File: python/copass-management/src/copass_management/tools/resume_trigger.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def resume_trigger(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    trigger_id = str(input["trigger_id"])
    trigger = await ctx.client.agents.triggers.update_by_id(
        ctx.sandbox_id, trigger_id, status="active",
    )
    return {"trigger": trigger}

File: typescript/packages/management/src/tools/resume_trigger.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const resumeTrigger: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const triggerId = String(input.trigger_id);
  const trigger = await ctx.client.agents.triggers.updateById(
    ctx.sandboxId,
    triggerId,
    { status: 'active' },
  );
  return { trigger };
};
